### PR TITLE
netty: allow setting eventloop group shutdown grace values

### PR DESCRIPTION
**Pull request by pyr**
addresses #638

### CODE REVIEW

**Comment by pyr** [comment]
Circle-CI is not configured hence the test issue, but tests run fine for me locally.

**Review by arnaudgeiser — COMMENTED** [review]
I'm obviously biased, but I would like those changes merged on Aleph as my company requires it.

File: src/aleph/netty.clj [hunk 1]
```diff
@@ -1219,82 +1227,95 @@ initialize an DnsAddressResolverGroup instance.
...
+                 socket-address epoll? default-shutdown-quiet-period
+                 default-shutdown-timeout))
+  ([pipeline-builder
+    ssl-context
+    bootstrap-transform
+    on-close
+    ^SocketAddress socket-address
+    epoll?
+    shutdown-quiet-period
+    shutdown-timeout]
```
[the remark this answers is not in the document]
Comment (KingMob) [reply src/aleph/netty.clj:1246]: +1 for collecting into a map. We have too many high-arity fns in Aleph, and I'd prefer not to add new ones.

**Comment by pyr** [comment]
I'll tackle the comments later today, thanks for the review!

File: src/aleph/netty.clj [hunk 2]
```diff
@@ -1219,82 +1227,95 @@ initialize an DnsAddressResolverGroup instance.
...
+    epoll?]
+   (start-server pipeline-builder ssl-context bootstrap-transform on-close
+                 socket-address epoll? default-shutdown-quiet-period
+                 default-shutdown-timeout))
+  ([pipeline-builder
+    ssl-context
+    bootstrap-transform
+    on-close
+    ^SocketAddress socket-address
+    epoll?
```
[the remark this answers is not in the document]
Comment (pyr) [reply src/aleph/netty.clj]: I added the two currently supported one: `:epoll`, and `:nio`

File: src/aleph/netty.clj (hunk 1 above)
[the remark this answers is not in the document]
Comment (pyr) [reply src/aleph/netty.clj]: done

File: src/aleph/http/server.clj [hunk 3]
```diff
@@ -577,12 +577,16 @@
            epoll?
            compression?
            continue-handler
-           continue-executor]
+           continue-executor
+           shutdown-quiet-period
+           shutdown-timeout]
```
[the remark this answers is not in the document]
Comment (pyr) [reply src/aleph/http/server.clj:582]: done

File: src/aleph/tcp.clj [hunk 4]
```diff
@@ -81,10 +81,13 @@
    | `pipeline-transform` | a function that takes an `io.netty.channel.ChannelPipeline` object, which represents a connection, and modifies it.
    | `raw-stream?` | if true, messages from the stream will be `io.netty.buffer.ByteBuf` objects rather than byte-arrays.  This will minimize copying, but means that care must be taken with Netty's buffer reference counting.  Only recommended for advanced users."
   [handler
-   {:keys [port socket-address ssl-context bootstrap-transform pipeline-transform epoll?]
+   {:keys [port socket-address ssl-context bootstrap-transform pipeline-transform epoll?
+           shutdown-quiet-period shutdown-timeout]
```
[the remark this answers is not in the document]
Comment (pyr) [reply src/aleph/tcp.clj:87]: done

**Review by arnaudgeiser — APPROVED** [review]
Thank you @pyr.

I let you press the button when you are ready @KingMob. :)

**Comment by KingMob** [comment]
I'm not sure we need to hold up the PR for the Circle issue. Erik doesn't recognize it. I can't tell where it's coming from. I guess we'll see.

Merging for now

**Pull request merged**